src/Sentry/Protocol/Span.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Sentry.Internal.Extensions;
+
+namespace Sentry.Protocol
+{
+    // https://develop.sentry.dev/sdk/event-payloads/span
+    public class Span : ISpan, IJsonSerializable
+    {
+        public SentryId SpanId { get; }
+        public SentryId? ParentSpanId { get; }
+        public SentryId TraceId { get; set; }
+        public DateTimeOffset StartTimestamp { get; set; }
+        public DateTimeOffset EndTimestamp { get; set; }
+        public string Operation { get; }
+        public string? Description { get; set; }
+        public SpanStatus? Status { get; set; }
+        public bool IsSampled { get; set; }
+
+        private ConcurrentDictionary<string, string>? _tags;
+        public IReadOnlyDictionary<string, string> Tags => _tags ??= new ConcurrentDictionary<string, string>();
+
+        private ConcurrentDictionary<string, object>? _data;
+        public IReadOnlyDictionary<string, object> Data => _data ??= new ConcurrentDictionary<string, object>();
+
+        internal Span(SentryId? spanId = null, SentryId? parentSpanId = null, string operation = "unknown")
+        {
+            SpanId = spanId ?? SentryId.Create();
+            ParentSpanId = parentSpanId;
+            TraceId = SentryId.Create();
+            StartTimestamp = EndTimestamp = DateTimeOffset.Now;
+            Operation = operation;
+        }
+
+        public ISpan StartChild(string operation) => new Span(null, SpanId, operation);
+
+        public void Finish(SpanStatus status = SpanStatus.Ok)
+        {
+            EndTimestamp = DateTimeOffset.Now;
+            Status = status;
+        }
+
+        public void WriteTo(Utf8JsonWriter writer)
+        {
+            writer.WriteStartObject();
+
+            writer.WriteSerializable("span_id", SpanId);
+
+            if (ParentSpanId is {} parentSpanId)
+            {
+                writer.WriteSerializable("parent_span_id", parentSpanId);
+            }
+
+            writer.WriteSerializable("trace_id", TraceId);
+            writer.WriteString("start_timestamp", StartTimestamp);
+            writer.WriteString("timestamp", EndTimestamp);
+
+            if (!string.IsNullOrWhiteSpace(Operation))
```

Comment:
This is mutable so it's racy. It could be non empty here but suddenly empty on the next line

Reply:
do you think the span should be frozen after finishing? or when serializing?

Reply:
It seems natural to me that once it's finished it should be frozen.

Curious what other SDKs did here.